doc/user/content/sql/create-source/mysql.md

```diff
@@ -0,0 +1,387 @@
+as `text`, and will thus not offer the expected MySQL type features. For
+example:
+
+* [`enum`]: the implicit ordering of the original MySQL `enum` type is not
+  preserved, as Materialize will sort values as `text`.
+
+##### Truncation
+
+Tables replicated into Materialize should not be truncated. If a table is
+truncated while replicated, the whole source becomes inaccessible and will not
+produce any data until it is recreated. Instead, remove all rows from a table
+using an unqualified `DELETE`.
+
+```sql
+DELETE FROM t;
+```
+
+## Examples
+
+{{< warning >}}
+Before creating a MySQL source, you must enable GTID-based binlog replication in the
+upstream database. For step-by-step instructions, see the integration guide for
+your MySQL service: [Amazon RDS](/ingest-data/mysql-amazon-rds/),
+[Amazon Aurora](/ingest-data/mysql-amazon-aurora/),
+[Azure DB](/ingest-data/mysql-azure-db/),
+[Google Cloud SQL](/ingest-data/mysql-google-cloud-sql/),
+[Self-hosted](/ingest-data/mysql-self-hosted/).
+{{< /warning >}}
+
+### Creating a connection
+
+A connection describes how to connect and authenticate to an external system you
+want Materialize to read data from.
+
+Once created, a connection is **reusable** across multiple `CREATE SOURCE`
+statements. For more details on creating connections, check the
+[`CREATE CONNECTION`](/sql/create-connection/#mysql) documentation page.
+
+```sql
+CREATE SECRET mysqlpass AS '<MYSQL_PASSWORD>';
+
+CREATE CONNECTION mysql_connection TO MYSQL (
+    HOST 'instance.foo000.us-west-1.rds.amazonaws.com',
+    PORT 3306,
+    USER 'root',
+    PASSWORD SECRET mysqlpass
+);
+```
+
+If your MySQL server is not exposed to the public internet, you can
+[tunnel the connection](/sql/create-connection/#network-security-connections)
+through an SSH bastion host.
+
+{{< tabs tabID="1" >}}
+{{< tab "SSH tunnel">}}
+```sql
+CREATE CONNECTION ssh_connection TO SSH TUNNEL (
+    HOST 'bastion-host',
+    PORT 22,
+    USER 'materialize',
+);
+```
+
+```sql
+CREATE CONNECTION mysql_connection TO MYSQL (
+    HOST 'instance.foo000.us-west-1.rds.amazonaws.com',
+    SSH TUNNEL ssh_connection,
+);
+```
+
+For step-by-step instructions on creating SSH tunnel connections and configuring
+an SSH bastion server to accept connections from Materialize, check
+[this guide](/ops/network-security/ssh-tunnel/).
+
+{{< /tab >}}
+{{< /tabs >}}
+
+### Creating a source {#create-source-example}
+
+_Create subsources for all tables in MySQL_
+
+```sql
+CREATE SOURCE mz_source
+    FROM MYSQL CONNECTION mysql_connection
+    FOR ALL TABLES;
+```
+
+_Create subsources for all tables from specific schemas in MySQL_
+
+```sql
+CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_connection
+  FOR SCHEMAS (mydb, project);
+```
+
+_Create subsources for specific tables in MySQL_
+
+```sql
+CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_connection
+  FOR TABLES (mydb.table_1, mydb.table_2 AS alias_table_2);
+```
+
+#### Handling unsupported types
```

Comment:
I kind of hate that we punt these to the `Examples`, but it's also how we structure the PostgreSQL page. Planning to bring these up to `Features` as a follow-up.